docs/adr/0005-feature-flags.md

```diff
@@ -0,0 +1,187 @@
+
+It's much easier to spot the zombie flags if our code is being maintained properly after feature complete. The if/elses that usually associate with feature flags should be deleted upon completition of a feature, and with a simple search, we should be able to quickly spot stale feature flags
+
+```
+From @Vinnl
+One thing that's been helpful on Relay is to define a union of strings enumerating all active feature flags in TypeScript. Then we can just remove a flag from that type, and TS will tell us which bits of code will need to be updated - because it's often not entire if/else branches you can just delete.
+```
+
+### Ownership
+Looking ahead, there may come a time when we need a system that enables the ability to assign ownership of a flag
+
+Feature flags offer a powerful way to directly modify the behavior of production systems, acting as a kill switch. The configuration of flags should undergo the same level of change control as the deployment of production code. Associating explicit ownership with a flag allows for the application of change control policies. For instance, individuals can be restricted from modifying flags they do not own, or approval can be required before making feature-flag configuration changes in the production environment.
+
+### Flag Tagging
+Attaching information like expiration dates and ownership to a flag are specific examples of a more general capability: the ability to tag flags.
+
+Tagging refers to the ability to attach arbitrary key:value pairs to a feature flag. This concept is employed in infrastructure systems like Kubernetes and AWS. It provides a highly adaptable method for users to associate semantic metadata with entities within the system. Tags can be used to track various aspects of a flag, such as its creation date, expiration date, life-cycle status, flag type, ownership, authorized modifier etc.
+
+## Considered Options
+* Environment Variables
+* JSON
+* [Unleash](https://www.getunleash.io/) Open Source Feature Flag (self host)
+* [Unleash](https://www.getunleash.io/) (managed cloud)
+
+## Decision Outcome
+
+A team decision has been made on Thursday, May 25. We will go with the `environment variable` approach similar to Relay. 
+* We will expose the relevant feature flag env vars via a runtime data endpoint similar to [relay](https://relay.firefox.com/api/v1/runtime_data). 
+* We will name the flags according to the convention listed above. 
+* There will be a boolean attached to each flag and there won't be additional metadata attached at this moment
+
+
+## Pros and Cons of the Options <!-- optional -->
+
+### Environment Variable
+
+```
+ENABLED_FEATURE_FLAGS = "premium_subplat_email, free_breaches_hibp"
+
+function isFlagEnabled(flag: string): boolean {
+  return (process.env.ENABLED_FEATURE_FLAGS ?? '').split(',').includes(flag);
+}
+
+const confettiScript = isFlagEnabled('party-mode')
+  ? '<script src="confetti.js"></script>'
+  : "";
+```
```

Comment:
Q: Does this allow us to do %-based rollouts?
Recently in FxA we did a 15% rollout of some new React work before rolling out to 100% of users.